nnef/src/ops/nnef/deser.rs

```diff
@@ -387,20 +387,31 @@ pub fn conv_or_deconv(
     builder.wire(op, &[input])
 }
 
+fn get_spatial_shape(shape: &[usize]) -> TractResult<TVec<usize>> {
```

Comment:
This logic feels a bit brittle. If I understand this right, the data format can actually be NCHW or... just HW. Where does this come from ? Is it operator dependant ? (like Conv VS pooling ?)

What if we have a 3D-HW ? This test may make the wrong decision... Can we have a more robust test to choose the format ? Like propagate the info to pool_spec_for_pools ?

Reply:
Yes that's a good point. Instead of overcomplicating the PR, I've just add a strict check enforcing the NNEF specification.